buildbot/buildbot/worker.py

```diff
@@ -102,9 +102,10 @@ def run(self, cmd, log_stdout=True, timeout=60, cwd='', **kwargs):
 
 
 @contextmanager
-def work(db, old_state, temp_state, done_state):
+def work_with_func(db, old_state, temp_state, done_state_func):
```

Comment:
I would like this a tiny bit more if this, instead of using a helper function, used a more Pythony way of letting the final argument be _either_ a state or a function that produces a state. Namely:

```py
def work(db, old_state, temp_state, done_state):
    # done_state is either a (constant) state or a function that
    # gives a final state for a given job.
    if isinstance(done_state, str):
        done_state = lambda _: done_state
```

Reply:
Fixed.